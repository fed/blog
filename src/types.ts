import { MarkdownRemark } from '../graphql-types';

export enum Category {
    BACONJS = 'baconjs',
    RXJS = 'rxjs',
    JAVASCRIPT = 'javascript',
    REACT = 'react',
    TESTING = 'testing',
    ACCESSIBILITY = 'accessibility',
    BROWSERS = 'browsers',
    GENERAL = 'general',
}

// export type Post = MarkdownRemark;

export interface Post {
    id: string;
    html: string;
    timeToRead: number;
    frontmatter: {
        title: string;
        date: string;
        spoiler: string;
        category: Category;
    };
    fields: {
        slug: string;
    };
}
