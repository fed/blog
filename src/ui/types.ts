export enum CategoryId {
    GENERAL = 'general',
    DOM = 'dom',
    JAVASCRIPT = 'javascript',
    TESTING = 'testing',
    ACCESSIBILITY = 'accessibility',
    DECLARATIVE = 'declarative',
    FRP = 'frp',
}

export interface Post {
    id: string;
    title: string;
    spoiler: string;
    date: string;
    categoryId: CategoryId;
    slug: string;
}
