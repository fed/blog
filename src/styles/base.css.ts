import { style } from '@vanilla-extract/css';

import { colors, fontFamilies, fontSizes, fontWeights, lineHeights, borderRadius } from './constants';

// The style definition (e.g. baseHeadingStyle) and its corresponding CSS declaration (e.g. baseHeadingStyleImpl)
// are declared separately because unfortunately when declaring global styles (as in the markdown.css.ts file)
// you need to pass in an object with CSS properties and not a vanilla-extract classname.

export const baseHeadingStyleImpl = {
    color: colors.grayDark,
    fontFamily: fontFamilies.sansSerif,
    lineHeight: lineHeights.md,
};

export const baseHeadingStyle = style(baseHeadingStyleImpl);

export const baseTitleStyle = style([
    baseHeadingStyle,
    {
        fontWeight: fontWeights.bold,
        margin: 0,
    },
]);

export const baseParagraphStyleImpl = {
    color: colors.grayDark,
    fontFamily: fontFamilies.sansSerif,
    fontSize: fontSizes.md,
    fontWeight: fontWeights.light,
    lineHeight: lineHeights.lg,
};

export const baseParagraphStyle = style(baseParagraphStyleImpl);

export const baseFocusStateStyle = style({
    selectors: {
        '&:focus': {
            outline: `${borderRadius.default} solid ${colors.pink}`,
            outlineOffset: '2px',
            borderRadius: '1px',
            borderBottomStyle: 'none',
        },
        '&:focus:hover': {
            borderBottomStyle: 'none',
        },
    },
});

export const baseLinkStyle = style([
    baseFocusStateStyle,
    {
        borderBottom: `1px solid ${colors.grayLight}`,
        color: colors.blue,
        paddingBottom: '1px',
        textDecoration: 'none',
        transition: 'border-bottom 0.5s ease',
        wordBreak: 'break-word',
        selectors: {
            '&:hover': {
                borderBottom: `1px solid ${colors.blue}`,
            },
        },
    },
]);

export const srOnlyStyle = style({
    borderWidth: 0,
    clip: 'rect(0, 0, 0, 0)',
    height: '1px',
    margin: '-1px',
    overflow: 'hidden',
    padding: 0,
    position: 'absolute',
    whiteSpace: 'nowrap',
    width: '1px',
});
