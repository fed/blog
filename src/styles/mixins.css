/* Colors */
@value color-gray-dark: #333;
@value color-gray-md: #8f8f8f;
@value color-gray-light: #e1e1e1;
@value color-blue: #056cc1;

/* Font families */
@value font-serif: 'Lora', serif;
@value font-sans-serif: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif;

/* Dimensions */
@value size-container-sm: 768px;
@value size-container-lg: 875px;

.title {
    color: color-gray-dark;
    font-family: font-sans-serif;
    font-size: 36px;
    font-weight: 700;
    line-height: 1.4;
    margin: 0;
}

.subtitle {
    color: color-gray-md;
    font-family: font-sans-serif;
    font-size: 28px;
    font-weight: 300;
    margin: 10px 0 0;
}

.content {
    color: color-gray-dark;
    font-family: font-serif;
    font-size: 21px;
    line-height: 1.6;
}

.link {
    border-bottom: 1px solid color-gray-light;
    color: color-blue;
    padding-bottom: 2px;
    text-decoration: none;
    transition: border-bottom 0.5s ease;
}

.link:hover {
    border-bottom: 1px solid color-blue;
}

.sansSerif {
    font-family: font-sans-serif;
}

.smallContainer {
    max-width: size-container-sm;
}

.largeContainer {
    max-width: size-container-lg;
}
