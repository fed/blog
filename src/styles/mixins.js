import { css } from 'styled-components';

/* Colors */
const colorGrayDark = '#333';
const colorGrayMedium = '#8f8f8f';
const colorGrayLight = '#e1e1e1';
const colorBlue = '#056cc1';
const colorGreen = '#3cca5d';
const colorYellow = '#f9ca5e';
const colorPink = '#dc2a5f';

/* Font families */
const fontFamilySerif = 'Lora, serif';
const fontFamilySansSerif =
    '-apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif';
const fontFamilyMonospace =
    '"SFMono-Medium", "SF Mono", "Segoe UI Mono", "Roboto Mono", "Ubuntu Mono", Menlo, Consolas, Courier, monospace';

/* Dimensions */
const sizeContainerSmall = '768px';
const sizeContainerLarge = '875px';

/* Reusable styles */
export const baseTitleStyles = css`
    color: ${colorGrayDark};
    font-family: ${fontFamilySansSerif};
    font-size: 36px;
    font-weight: 700;
    line-height: 1.4;
    margin: 0;
`;

export const baseSubtitleStyles = css`
    color: ${colorGrayDark};
    font-family: ${fontFamilySansSerif};
    font-size: 18px;
    font-weight: 300;
`;

export const baseParagraphStyles = css`
    color: ${colorGrayDark};
    font-family: ${fontFamilySerif};
    font-size: 21px;
    line-height: 1.6;
`;

export const baseRawLinkStyles = css`
    color: ${colorBlue};
    text-decoration: none;
    transition: border-bottom 0.5s ease;
    word-break: break-word;
`;

export const baseFocusStateStyles = css`
    :focus {
        border-bottom-style: none;
        border-radius: 3px;
        box-shadow: 0 0 0 3px color-pink;
        outline: none;
    }
`;

export const baseLinkStyles = css`
    ${baseRawLinkStyles};
    border-bottom: 1px solid ${colorGrayLight};
    padding-bottom: 2px;
    :hover {
        border-bottom: 1px solid ${colorBlue};
    }
`;

export const baseSansSerifStyles = css`
    font-family: ${fontFamilySansSerif};
`;

export const baseSmallContainerStyles = css`
    max-width: ${sizeContainerSmall};
`;

export const baseLargeContainerStyles = css`
    max-width: ${sizeContainerLarge};
`;
