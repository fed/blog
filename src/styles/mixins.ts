import { css } from 'styled-components';

/* Colors */
export const colorGrayDark = '#1f2937';
export const colorGrayMedium = '#6a7482';
export const colorGrayLight = '#e1e1e1';
export const colorBlue = '#056cc1';
export const colorGreen = '#3cca5d';
export const colorYellow = '#f9ca5e';
export const colorPink = '#dc2a5f';

/* Font families */
export const fontFamilySansSerif =
    '-apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif';
export const fontFamilyMonospace =
    '"SFMono-Medium", "SF Mono", "Segoe UI Mono", "Roboto Mono", "Ubuntu Mono", Menlo, Consolas, Courier, monospace';

/* Dimensions */
export const sizeContainerExtraSmall = '480px';
export const sizeContainerSmall = '768px';
export const sizeContainerMedium = '875px';
export const sizeContainerLarge = '1024px';
export const sizeContainerExtraLarge = '1290px';

/* Reusable styles */
export const baseHeadingStyles = css`
    color: ${colorGrayDark};
    font-family: ${fontFamilySansSerif};
    line-height: 1.4;
`;

export const baseTitleStyles = css`
    ${baseHeadingStyles};
    font-size: 30px;
    font-weight: 700;
    margin: 0;
`;

export const baseParagraphStyles = css`
    color: ${colorGrayDark};
    font-family: ${fontFamilySansSerif};
    font-size: 18px;
    font-weight: 250;
    line-height: 1.6;
`;

export const baseFocusStateStyles = css`
    :focus {
        border-bottom-style: none;
        border-radius: 3px;
        box-shadow: 0 0 0 3px ${colorPink};
        outline: none;
    }
`;

export const baseLinkStyles = css`
    ${baseFocusStateStyles};
    border-bottom: 1px solid ${colorGrayLight};
    color: ${colorBlue};
    padding-bottom: 2px;
    text-decoration: none;
    transition: border-bottom 0.5s ease;
    word-break: break-word;
    :hover {
        border-bottom: 1px solid ${colorBlue};
    }
`;
