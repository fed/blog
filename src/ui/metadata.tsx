import React from 'react';
import styled from 'styled-components';

import { colors, fontFamilies, gridSize } from '../styles/constants';
import { Category } from './category';
import { CategoryId } from './types';

const Container = styled.div`
    margin: 0 0 ${0.625 * gridSize}px;
`;

const PublicationDate = styled.span`
    color: ${colors.grayMedium};
    font-family: ${fontFamilies.sansSerif};
    font-size: 12px;
    margin-right: ${1.25 * gridSize}px;
    text-transform: uppercase;
`;

interface Props {
    date: string;
    categoryId: CategoryId;
}

export const Metadata: React.FC<Props> = ({ date, categoryId }) => {
    if (!date && !categoryId) {
        return null;
    }

    return (
        <Container>
            {date ? <PublicationDate data-testid="metadata-publication-date">{date}</PublicationDate> : null}
            {categoryId ? <Category categoryId={categoryId} /> : null}
        </Container>
    );
};
