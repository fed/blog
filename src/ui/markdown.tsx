import React from 'react';
import styled from 'styled-components';

import { colors, fontFamilies, fontWeights, gridSize } from '../styles/constants';
import { baseHeadingStyles, baseLinkStyles, baseParagraphStyles } from '../styles/mixins';

const Container = styled.div`
    p,
    ul,
    ol {
        ${baseParagraphStyles};
    }

    a:not(.gatsby-resp-image-link) {
        ${baseLinkStyles};
    }

    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
        ${baseHeadingStyles};
        font-weight: ${fontWeights.bold};
        margin: ${6.25 * gridSize}px 0 0;
    }

    b,
    strong {
        font-weight: ${fontWeights.bold};
    }

    img {
        max-width: 100%;
    }

    blockquote {
        background-color: rgb(222, 235, 255);
        border-radius: 3px;
        margin-left: 0;
        margin-right: 0;
        padding: ${gridSize}px ${2 * gridSize}px;

        p {
            font-size: 16px;
            font-weight: ${fontWeights.normal};
            margin: 0;
        }

        a:not(.gatsby-resp-image-link) {
            color: ${colors.grayDark};
            transition: none;

            &,
            :hover {
                border-bottom: 2px solid ${colors.blue};
            }
        }
    }

    /* Inline code */
    code:not([class='grvsc-code']) {
        background-color: rgba(27, 31, 35, 0.05);
        border-radius: 3px;
        font-size: 80%;
        margin: 0;
        padding: ${0.375 * gridSize}px ${0.75 * gridSize}px;
        white-space: normal;
        font-weight: ${fontWeights.normal};
    }

    /* Code blocks */
    pre.grvsc-container {
        line-height: 1.75;
    }

    pre,
    code,
    code,
    kbd,
    samp {
        font-family: ${fontFamilies.monospace};
    }

    /* Add some vertical spacing to all images */
    .gatsby-resp-image-figure,
    p > .gatsby-resp-image-wrapper {
        margin-bottom: ${6 * gridSize}px;
        margin-top: ${6 * gridSize}px;
    }

    .gatsby-resp-image-figcaption {
        ${baseParagraphStyles};
        font-size: 12px;
        margin-top: ${gridSize}px;
        text-align: center;
    }
`;

export const Markdown: React.FC = ({ children }) => <Container dangerouslySetInnerHTML={{ __html: children }} />;
