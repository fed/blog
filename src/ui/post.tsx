import React from 'react';
import styled from 'styled-components';

import { baseTitleStyles } from '../styles/mixins';
import { Markdown } from './markdown';
import { Metadata } from './metadata';
import { CategoryId } from './types';

const Title = styled.h1`
    ${baseTitleStyles};
`;

interface Props {
    title: string;
    date: string;
    categoryId: CategoryId;
    body: string;
}

export const Post: React.FC<Props> = ({ title, date, categoryId, body }) => (
    <article>
        <Metadata date={date} categoryId={categoryId} />
        <Title>{title}</Title>
        <Markdown>{body}</Markdown>
    </article>
);
