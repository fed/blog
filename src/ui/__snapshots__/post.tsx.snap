// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post renders correctly 1`] = `
<article>
  <header
    className="sc-hKwCoD hYmsuj"
  >
    <h1
      className="sc-eCImvq kavgNB"
    >
      
    </h1>
    <div
      className="sc-jRQAMF kBxSTL"
    >
      <ul
        className="sc-bdvvaa egwCKV"
      >
        <li
          className="sc-gsDJrp kIyLNX"
        >
          <img
            alt="Date published"
            className="sc-dkPtyc cirgGZ"
            src="test-file-stub"
            title="Date published"
          />
          <time
            pubdate="pubdate"
          >
            October 19, 2019
          </time>
        </li>
        <li
          className="sc-gsDJrp kIyLNX"
        >
          <img
            alt="Category"
            className="sc-dkPtyc cirgGZ"
            src="test-file-stub"
            title="Category"
          />
          General
        </li>
        <li
          className="sc-gsDJrp kIyLNX"
        >
          <img
            alt="Time to read"
            className="sc-dkPtyc cirgGZ"
            src="test-file-stub"
            title="Time to read"
          />
          2
           min. read
        </li>
      </ul>
    </div>
  </header>
  <div
    className="sc-gKckTs hhaZVw"
    dangerouslySetInnerHTML={
      Object {
        "__html": "This is a blog post",
      }
    }
  />
</article>
`;
